data/data.go

```diff
@@ -31,34 +31,39 @@ type EffxYaml struct {
 	FilePath string
 }
 
-// prevents us from overwriting tags and annotations that the customer
-// has already filled in.
-func safelySetField(configMap *map[string]string, field, value string) {
-	// if field is already set, do not overwrite
-	if _, ok := (*configMap)[field]; ok {
-		return
+func setMetadata(config *effx_api.ConfigurationFile, m *metadata.Result) *effx_api.ConfigurationFile {
+	if m == nil {
+		return config
 	}
 
-	// if field is not set, set it
-	(*configMap)[field] = value
-}
+	var inferredTags []string
+	tags, ok := config.GetTagsOk()
+	if !ok {
+		tags = &map[string]string{}
+	}
 
-func setMetadata(config *effx_api.ConfigurationFile, m *metadata.Result) *effx_api.ConfigurationFile {
-	if m != nil {
-		if config.Annotations == nil {
-			config.Annotations = &map[string]string{}
-		}
-		if config.Tags == nil {
-			config.Tags = &map[string]string{}
-		}
+	annotations, ok := config.GetAnnotationsOk()
+	if !ok {
+		annotations = &map[string]string{}
+	}
 
-		if m.Language != "" && m.Version != "" {
-			safelySetField(config.Annotations, "effx.io/inferred-tags", fmt.Sprintf("language,%s", m.Language))
-			safelySetField(config.Tags, "language", strings.ToLower(m.Language))
-			safelySetField(config.Tags, m.Language, strings.ToLower(m.Version))
-		}
+	if m.Language != "" {
+		languageTag := "language"
+		(*tags)[languageTag] = m.Language
+		inferredTags = append(inferredTags, languageTag)
 	}
 
+	if m.Language != "" && m.Version != "" {
```

Comment:
what if the users have a tag set for language.

Reply:
All of that logic will happen in the API. Since we don't parse the contents of YAMLs in this client, inferred tags will get appended to objects server-side.

We can expect the API to omit inferred tags when the tag was already set explicitly.